cpp/src/compaction/stream_compaction_ops.cu

```diff
@@ -224,109 +230,232 @@ gdf_error gpu_apply_stencil(gdf_column *lhs, gdf_column * stencil, gdf_column *
-	//whoever calls that should handle that
-	if(width == 1){
-		thrust::detail::normal_iterator<thrust::device_ptr<int8_t> > input_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int8_t *) lhs->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int8_t> > output_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int8_t *) output->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int8_t> > output_end =
-				thrust::copy_if(exec,input_start,input_start + lhs->size,zipped_stencil_iter,output_start,is_stencil_true<thrust::detail::normal_iterator<thrust::device_ptr<int8_t> >::value_type >());
-		output->size = output_end - output_start;
-	}else if(width == 2){
-		thrust::detail::normal_iterator<thrust::device_ptr<int16_t> > input_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int16_t *) lhs->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int16_t> > output_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int16_t *) output->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int16_t> > output_end =
-				thrust::copy_if(exec,input_start,input_start + lhs->size,zipped_stencil_iter,output_start,is_stencil_true<thrust::detail::normal_iterator<thrust::device_ptr<int8_t> >::value_type >());
-		output->size = output_end - output_start;
-	}else if(width == 4){
-		thrust::detail::normal_iterator<thrust::device_ptr<int32_t> > input_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int32_t *) lhs->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int32_t> > output_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int32_t *) output->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int32_t> > output_end =
-				thrust::copy_if(exec,input_start,input_start + lhs->size,zipped_stencil_iter,output_start,is_stencil_true<thrust::detail::normal_iterator<thrust::device_ptr<int8_t> >::value_type >());
-		output->size = output_end - output_start;
-	}else if(width == 8){
-		thrust::detail::normal_iterator<thrust::device_ptr<int64_t> > input_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int64_t *) lhs->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int64_t> > output_start =
-				thrust::detail::make_normal_iterator(thrust::device_pointer_cast((int64_t *) output->data));
-		thrust::detail::normal_iterator<thrust::device_ptr<int64_t> > output_end =
-				thrust::copy_if(exec,input_start,input_start + lhs->size,zipped_stencil_iter,output_start,is_stencil_true<thrust::detail::normal_iterator<thrust::device_ptr<int8_t> >::value_type >());
-		output->size = output_end - output_start;
-	}
+		//NOTE!!!! the output column is getting set to a specific size  but we are NOT compacting the allocation,
+		//whoever calls that should handle that
+		if(width == 1){
```

Comment:
This should use the `type_dispatcher` instead of this manual `if/else` on the bit-width.

https://github.com/rapidsai/cudf/blob/master/cpp/src/dataframe/type_dispatcher.hpp#L89

Reply:
Done.